packages/flutter_cubit/lib/src/cubit_provider.dart

```diff
@@ -128,8 +128,17 @@ class CubitProvider<T extends CubitStream<Object>>
     return InheritedProvider<T>(
       create: _create,
       dispose: _dispose,
-      child: child,
-      lazy: lazy,
+      child: DeferredInheritedProvider<T, S>(
```

Comment:
Was it intentional to nest a `DeferredInheritedProvider` within an `InheritedProvider`? Can't the `InheritedProvider` be replaced with the `DeferredInheritedProvider` instead?

Reply:
From my understanding `DeferredInheritedProvider` only exposes the object(s) emitted but not the actual one listened to. So in this case `DeferredInheritedProvider` only exposes the `state` but not the `Cubit`. However, I only have a superficial understanding of the various providers so there might be a better way to achieve the same.